scratchwork/main.go

```diff
@@ -59,27 +95,62 @@ func getDeckList() (StaticDeckList, error) {
 	return sdl, nil
 }
 
+func getCardPos() (PositionalRectangles, error) {
+	cardPos := PositionalRectangles{}
+
+	endpoint := fmt.Sprintf("%s/positional-rectangles", legendHost)
+	resp, err := http.Get(endpoint)
+	if err != nil {
+		return cardPos, errors.Wrap(err, "getting card positions failed")
+	}
+
+	dec := json.NewDecoder(resp.Body)
+	err = dec.Decode(&cardPos)
+	if err != nil {
+		return cardPos, errors.Wrap(err, "response could not be interpreted as card positions")
+	}
+	return cardPos, nil
+}
+
+func convertDeckListCodesToNames(list StaticDeckList) StaticDeckList{
+	deckTracker := make(map[string]int)
+	for key, value := range list.CardsInDeck {
+		cardName := cardCodeToName[key]
+		deckTracker[cardName] = value
+	}
+	list.CardsInDeck = deckTracker
+	return list
+}
+
+
 type mainLoopArgs struct {
 }
 
 func mainloop(args mainLoopArgs) {
 	var (
 		signalUpdateDecklist = time.NewTicker(4 * time.Second)
 		currentDeckList      = StaticDeckList{}
+		currentCardPositions = PositionalRectangles{}
 	)
-
+	
 	for {
 		select {
 		case <-signalUpdateDecklist.C:
-			newdecklist, err := getDeckList()
+			newDeckList, err := getDeckList()
 			if err != nil {
 				// %+v prints error with the back trace
-				fmt.Println("encountered error fetching new decklist value:\n", err)
+				fmt.Println("encountered error fetching new decklist value:", err)
 			} else {
-				currentDeckList = newdecklist
+				currentDeckList = convertDeckListCodesToNames(newDeckList)
 				spew.Dump(currentDeckList)
 			}
-
+			newCardPos, err := getCardPos()
```

Comment:
This should have it's own ticker -- I think

Reply:
1. they're different functionalities, so they should have separate ticker
2. they also operate on different time frames: we need to getCardPos regularly, therefore, we shouldn't include it under the ticker of something like signalUpdateDecklist, which really only needs to be run once every like 5 minutes y'know?
3. the signalUpdateDecklist ticker, probably shouldn't exist at all: we can tell when we need to update the deck by checking gameState change.